src/source_sampling.cpp

```diff
@@ -254,39 +273,51 @@ void pyne::Sampler::mesh_tag_data(moab::Range ves,
   // That's okay. That's what the next line is all about:
   num_e_groups = num_groups(src_tag);
 
-  // Set the default value of max_num_cells to 1, so that the normal r2s and sub-voxel
-  // r2s can use the same form of pdf size description
+  // Set the default value of max_num_cells to 1, so that the structured mesh
+  // and unstructured mesh r2s can use the same form of pdf size description.
   max_num_cells = 1;
-  // set the default value of cell_fracs to 1.0
-  cell_fracs.resize(num_ves*max_num_cells);
-  for(int i=0; i<cell_fracs.size(); i++){
-    cell_fracs[i] = 1.0;
-  }
+  p_src_num_cells = 1;
+  int tag_size;
+  // set the default value of cell_number to -1, cell_fracs to 1.0.
+  cell_number.resize(num_ves, -1);
+  cell_fracs.resize(num_ves, 1.0);
   if (ve_type == moab::MBHEX) {
       // Read the cell_number tag and cell_fracs tag
       rval = mesh->tag_get_handle(cell_number_tag_name.c_str(),
                                   cell_number_tag);
       rval = mesh->tag_get_handle(cell_fracs_tag_name.c_str(),
                                   cell_fracs_tag);
-      max_num_cells = num_groups(cell_fracs_tag);
-      num_e_groups /= max_num_cells;
-      cell_fracs.resize(num_ves*max_num_cells);
-      rval = mesh->tag_get_data(cell_fracs_tag, ves, &cell_fracs[0]);
-      cell_number.resize(num_ves*max_num_cells);
-      rval = mesh->tag_get_data(cell_number_tag, ves, &cell_number[0]);
+      // has_cell_fracs is assinged in get_max_num_cells
```

Comment:
```suggestion
      // has_cell_fracs is assigned in get_max_num_cells
```

Reply:
I'm not sure I like the idea of `has_cell_fracs` being assigned without being anywhere in the function interface.  This is something that will easily be forgotten at some point.

Is there some step that is more like initialization/file reading when we can set this flag?